test/integration/login_test.go

```diff
@@ -0,0 +1,152 @@
+// +build integration,!no-etcd
+
+package integration
+
+import (
+	"os"
+	"strings"
+	"testing"
+
+	kclient "github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/openshift/origin/pkg/client"
+	"github.com/openshift/origin/pkg/cmd/cli/cmd"
+	newproject "github.com/openshift/origin/pkg/cmd/experimental/project"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	"github.com/openshift/origin/pkg/user/api"
+	"github.com/spf13/pflag"
+)
+
+func init() {
+	requireEtcd()
+}
+
+func TestLogin(t *testing.T) {
+	startConfig, err := StartTestMaster()
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	openshiftClient, openshiftClientConfig, err := startConfig.GetOpenshiftClient()
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	// empty config, should display message
+	loginOptions := newLoginOptions("", "", "", "", false)
+	err = loginOptions.GatherServerInfo()
+	if err == nil {
+		t.Fatalf("Raw login should error out")
+	} else {
+		if !strings.Contains(err.Error(), "OpenShift is not configured") {
+			t.Fatalf("Raw login should display OpenShift not configured message")
+		}
+	}
+
+	username := "joe"
+	password := "pass"
+	project := "the-singularity-is-near"
+	server := openshiftClientConfig.Host
+
+	loginOptions = newLoginOptions(server, username, password, "", true)
+
+	if err = loginOptions.GatherServerInfo(); err != nil {
+		t.Fatalf("Error trying to determine server info: ", err)
+	}
+
+	if err = loginOptions.GatherAuthInfo(); err != nil {
+		t.Fatalf("Error trying to determine auth info: ", err)
+	}
+
+	me, err := whoami(loginOptions.Config)
+	if err != nil {
+		t.Fatalf("unexpected error: ", err)
```

Comment:
try to avoid fatals in tests if it is possible for other subsequent parts of the test to pass.  In this case, doing the entire run could still succeed.